lib/ansible/modules/monitoring/grafana_datasource.py

```diff
@@ -0,0 +1,469 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017, Thierry Sallé (@tsalle)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.1'
+}
+
+DOCUMENTATION = '''
+---
+module: grafana_datasource
+author:
+  - "Thierry Sallé (@tsalle)"
+version_added: "2.5"
+short_description: Manage grafana datasources
+description:
+  - Create/update/delete grafana datasources via API
+options:
+  grafana_url:
+    required: true
+    description:
+      - Grafana url
+  name:
+    required: true
+    description:
+      - Name of the datasource
+  ds_type:
+    required: true
+    choices: [graphite,prometheus,elasticsearch,influxdb,opentsdb]
+    description:
+     - Type of the datasource C(graphite),C(prometheus),C(elasticsearch),C(influxdb),C(opentsdb)
+  url:
+    required: true
+    description:
+      - Url of the datasource
+  access:
+    required: false
+    default: proxy
+    choices: [proxy,direct]
+    description:
+      - Access mode for this datasource
+  grafana_user:
+    required: false
+    default: admin
+    description:
+      - Grafana API user
+  grafana_password:
+    required: false
+    default: admin
+    description:
+      - Grafana API password
+  grafana_api_key:
+    required: false
+    description:
+      - Grafana API key
+      - If set, grafana_user and grafana_password will be ignored
+  database:
+    required: false
+    description:
+      - Name of the database for the datasource.
+      - This options is required when the ds_type is influxdb or elasticsearch (index name)
+  user:
+    required: false
+    description:
+      - Datasource login user for influxdb datasources
+  password:
+    required: false
+    description:
+      - Datasource password
+  basic_auth_user:
+    required: false
+    description:
+      - Datasource basic auth user.
+      - Setting this option with basic_auth_password will enable basic auth.
+  basic_auth_password:
+    required: false
+    description:
+      - Datasource basic auth password, when basic auth is true
+  with_credentials:
+    required: false
+    default: false
+    description:
+      - Whether credentials such as cookies or auth headers should be sent with cross-site requests.
+  tls_client_cert:
+    required: false
+    description:
+      - client tls certificate.
+      - If tls_client_cert and tls_client_key are set, this will enable tls auth.
+      - begins with ----- BEGIN CERTIFICATE -----
+  tls_client_key:
+    required: false
+    description:
+      - client tls private key
+      - befins with ----- BEGIN RSA PRIVATE KEY -----
+  tls_ca_cert:
+    required: false
+    description:
+      - tls ca certificate for self signed certificates.
+      - only used when tls_client_cert and tls_client_key are set
```

Comment:
required_together?

Reply:
the tls_ca_cert is not mandatory when tls_client auth is enabled.
It's just an other option to trust the certificate of the datasource.